freqtrade/exchange/exchange.py

```diff
@@ -1983,6 +1982,126 @@ def get_funding_fees(
         else:
             return 0.0
 
+    @retrier
+    def get_liquidation_price(
+        self,
+        pair: str,
+        # Dry-run
+        open_rate: Optional[float] = None,   # Entry price of position
+        is_short: Optional[bool] = None,
+        position: Optional[float] = None,  # Absolute value of position size
+        wallet_balance: Optional[float] = None,  # Or margin balance
+        mm_ex_1: float = 0.0,  # (Binance) Cross only
+        upnl_ex_1: float = 0.0,  # (Binance) Cross only
+    ):
+        """
+        Set's the margin mode on the exchange to cross or isolated for a specific pair
+        :param pair: base/quote currency pair (e.g. "ADA/USDT")
+        """
+        if self.trading_mode == TradingMode.SPOT:
+            return None
+        elif (self.collateral is None):
+            raise OperationalException(f'{self.name}.collateral must be set for liquidation_price')
+        elif (self.trading_mode != TradingMode.FUTURES and self.collateral != Collateral.ISOLATED):
+            raise OperationalException(
+                f"{self.name} does not support {self.collateral.value} {self.trading_mode.value}")
+
+        if self._config['dry_run'] or not self.exchange_has("fetchPositions"):
+            if (
+                open_rate is None or
+                is_short is None or
+                position is None or
+                wallet_balance is None
+            ):
+                raise OperationalException(
+                    f"Parameters open_rate, is_short, position, wallet_balance are"
+                    f"required by {self.name}.liquidation_price for dry_run"
+                )
+
+            return self.dry_run_liquidation_price(
+                pair=pair,
+                open_rate=open_rate,
+                is_short=is_short,
+                position=position,
+                wallet_balance=wallet_balance,
+                mm_ex_1=mm_ex_1,
+                upnl_ex_1=upnl_ex_1
+            )
+
+        try:
+            positions = self._api.fetch_positions([pair])
+            pos = positions[0]
```

Comment:
i think we must check the length of "positions" before using it this way.

if you call this on a pair that has no positions open (or the exchange provides this empty for whatever reason), then `positions[0]` will yield a KeyError exception.